pkg/store/kotsstore/supportbundle_store.go

```diff
@@ -275,54 +228,58 @@ func (s *KOTSStore) GetSupportBundle(id string) (*types.SupportBundle, error) {
 		return nil, errors.Wrap(err, "failed to unmarshal")
 	}
 
-	treeindex, err := s.getSupportBundleMetafile(id, "treeindex")
-	if err != nil {
-		if s.IsNotFound(err) {
-			return &supportBundle, nil
+	if supportBundle.Status == "uploaded" {
+		treeindex, err := s.getSupportBundleMetafile(id, "treeindex")
+		if err != nil {
+			if s.IsNotFound(err) {
+				return &supportBundle, nil
+			}
+			return nil, errors.Wrap(err, "failed to get treeindex from s3")
 		}
-		return nil, errors.Wrap(err, "failed to get treeindex from s3")
+		supportBundle.TreeIndex = string(treeindex)
 	}
 
-	supportBundle.TreeIndex = string(treeindex)
-
 	return &supportBundle, nil
 }
 
-func (s *KOTSStore) CreatePendingSupportBundle(id string, appID string, clusterID string) error {
-	pendingSupportBundle := types.PendingSupportBundle{
-		ID:        id,
-		AppID:     appID,
-		ClusterID: clusterID,
-	}
-	b, err := json.Marshal(pendingSupportBundle)
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal")
-	}
+func (s *KOTSStore) CreateInProgressSupportBundle(supportBundle *types.SupportBundle) error {
+	id := supportBundle.ID
+	appID := supportBundle.AppID
 
-	clientset, err := k8sutil.GetClientset()
+	supportBundle.Status = "inProgress"
```

Comment:
Not a fan of camel case.  How about: "running", "failed", "complete".  And this is nit picky, but how about turning this into an enum type so we don't have these statuses multiply randomly.